frontend/src/pages/EditRoom/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { FunctionComponent, useState } from 'react';
+import { Checkbox, Col, Divider, Form, Input, Row, Button, Space, Alert, Spin } from 'antd';
+import {
+  RadarChartOutlined,
+  CloseOutlined,
+  CheckOutlined,
+} from '@ant-design/icons';
+import { useHistory } from 'react-router-dom';
+import { useRooms } from '../../services/rooms';
+import { Acknowledgment } from '../../services/acknowledgment';
+
+type EditRoomPageProps = {
+  roomId?: number;
+}
+
+const EditRoomPage: FunctionComponent<EditRoomPageProps> = ({
+  roomId
+}) => {
+  const history = useHistory();
+  const { rooms, createRoom, updateRoom } = useRooms();
+  const currentRoom = rooms?.find(room => room.id === roomId);
+
+  const [saving, setSaving] = useState(false);
+  const [saveErrors, setSaveErrors] = useState<string[]>();
+
+  const save = (values: { name: string, nodes: string[] }) => {
+    setSaving(true);
+    if (currentRoom) {
+      updateRoom({ id: currentRoom.id, name: values.name }).then(() => {
+        setSaving(false);
+        setSaveErrors(undefined);
+      }).catch((ack: Acknowledgment) => {
+        setSaving(false);
+        setSaveErrors(ack.errors);
+      })
+    } else {
+      createRoom({ name: values.name }).then(ack => {
+        setSaving(false);
+        setSaveErrors(undefined);
+        history.push(`/rooms/${ack.createdId}/edit`);
+      }).catch((ack: Acknowledgment) => {
+        setSaving(false);
+        setSaveErrors(ack.errors)
+      });
+    }
+  }
+
+  const startCalibration = () => {
+
+  }
+
+  return (
+    <Row justify="center">
+      <Col xl={8} lg={12} md={16} xs={24}>
+        {saveErrors?.map(error => (
+          <Alert message={error} type="error" />
```

Comment:
I get a warning here in the console that the `key` property is missing. Can you add it? I think just the error string should do it, there won't be duplicates in it.